erpnext/projects/doctype/timesheet/timesheet.js

```diff
@@ -20,33 +20,37 @@ frappe.ui.form.on("Timesheet", {
 					'project': child.project,
 					'status': ["!=", "Cancelled"]
 				}
-			}
-		}
+			};
+		};
 
 		frm.fields_dict['time_logs'].grid.get_field('project').get_query = function() {
 			return{
 				filters: {
 					'company': frm.doc.company
 				}
-			}
-		}
+			};
+		};
 	},
 
 	onload: function(frm){
 		if (frm.doc.__islocal && frm.doc.time_logs) {
 			calculate_time_and_amount(frm);
 		}
+
+		if (frm.doc.docstatus < 1) {
```

Comment:
It should only set employee and company if document is unsaved.

`if(frm.is_new())`